### PR TITLE
Allow loading PGN games using the digit 0 for castling

### LINKED ISSUE #157: Allow loading PGN games using the digit zero for castling
At the present moment, only PGN games using the uppercase letter O (O-O and O-O-O) for castling are considered to be valid; those using the FIDE standard won't pass the validation process.

According to [Wikipedia](https://en.wikipedia.org/wiki/Algebraic_notation_(chess)):

> Castling is indicated by the special notations 0-0 (for kingside castling) and 0-0-0 (queenside castling). While the FIDE standard is to use the digit zero (0-0 and 0-0-0), PGN uses the uppercase letter O (O-O and O-O-O).

For example, at this moment the [2021 FIDE World Championship](https://en.wikipedia.org/wiki/World_Chess_Championship_2021) games are not passing the validation because of this slight difference between the FIDE standard and the PGN standard.

out
**Figure 1**. This game could be loaded because the castling zeros were replaced with the letter "O".

Chess games using the FIDE standard for castling should also pass the PGN validation in the [`Chess\PGN\Validate`](https://github.com/chesslablab/php-chess/blob/master/src/PGN/Validate.php) file.

Keep it up, and happy learning and coding!

### PR DESCRIPTION
**Pull request by nicolafan**
Closes #157 
This simple fix should work.

### CODE REVIEW

**Comment by programarivm** [comment]
Thanks for sending this PR.

Let's review it and practice some unit testing skills. When fixing a bug or implementing a new feature, the following output should be obtained most of the times when running the tests.

```text
$ vendor/bin/phpunit tests
PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

...............................................................  63 / 483 ( 13%)
............................................................... 126 / 483 ( 26%)
............................................................... 189 / 483 ( 39%)
............................................................... 252 / 483 ( 52%)
............................................................... 315 / 483 ( 65%)
............................................................... 378 / 483 ( 78%)
............................................................... 441 / 483 ( 91%)
..........................................                      483 / 483 (100%)

Time: 00:57.583, Memory: 54.50 MB

OK (483 tests, 9917 assertions)
```

However, at this moment the tests are failing which probably points out something is to be refactored yet.

```text
$ vendor/bin/phpunit tests
PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

EEEEEEEEEEEEFEEEEEEEEEEEEEEEEEEEEEEEEEEEE.EEEEEEE.EEFEEEE.E.E.E  63 / 484 ( 13%)
EFEEEEEEEEEEEEEFEEFE.EEEEEEEEEEEEEEF...E.EEE................... 126 / 484 ( 26%)
.E.....................EEEEEEEEEE.E................E........EE. 189 / 484 ( 39%)
...E..............F......EE......E............................. 252 / 484 ( 52%)
.............................E................................. 315 / 484 ( 65%)
...........................F................EE................. 378 / 484 ( 78%)
..................EE...................................F....... 441 / 484 ( 91%)
...........................................                     484 / 484 (100%)

Time: 00:50.210, Memory: 56.50 MB

There were 115 errors:

1) Chess\Tests\Functional\GameTest::play_sample_games with data set #0 ('01.pgn')
Undefined index: O-O

/home/standard/projects/php-chess/src/Piece/King.php:59
/home/standard/projects/php-chess/src/Board.php:570
/home/standard/projects/php-chess/src/Board.php:817
/home/standard/projects/php-chess/src/Board.php:527
/home/standard/projects/php-chess/src/Board.php:559
/home/standard/projects/php-chess/src/Game.php:246
/home/standard/projects/php-chess/tests/functional/GameTest.php:29

...

9) Chess\Tests\Unit\PGN\ConvertTest::O_O_O
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
     'pgn' => 'O-O-O'
     'isCapture' => false
     'isCheck' => false
-    'type' => 'O-O|0-0[\+\#]{0,1}'
+    'type' => 'O-O-O|0-0-0[\+\#]{0,1}'
     'color' => 'w'
     'identity' => 'K'
     'position' => stdClass Object (
         'current' => 'e1'
-        'next' => 'g1'
+        'next' => 'c1'
     )
 )

/home/standard/projects/php-chess/tests/unit/PGN/ConvertTest.php:314

ERRORS!
Tests: 484, Assertions: 2365, Errors: 115, Failures: 9.
```

Keep it up, and happy learning and coding.

File: src/PGN/Symbol.php [hunk 1]
```diff
@@ -20,8 +20,8 @@ class Symbol
     const QUEEN = 'Q';
     const ROOK = 'R';
 
-    const CASTLING_SHORT = 'O-O';
```
Comment (programarivm) [inline src/PGN/Symbol.php:23]: What about adding two new constants called `CASTLING_SHORT_FIDE` and `CASTLING_LONG_FIDE`?

**Comment by nicolafan** [comment]
Yep! I ran only the unit test initially... So it seems that `PGN/Symbol.php` uses its data not only for validating imported games.

File: src/PGN/Symbol.php (hunk 1 above)
Comment (nicolafan) [reply src/PGN/Symbol.php:23]: This was actually my first thought when I started refactoring this.

Comment (nicolafan) [reply src/PGN/Symbol.php:23]: My ideas atm are trying to do this and separating the constants or adding a single fix directly into the validation.

**Comment by programarivm** [comment]
Probably this issue may require to change a few files actually, as well as adding some tests to check out new examples using FIDE notation. All sample data is using PGN notation atm.

**Comment by nicolafan** [comment]
It got closed automatically after I force-reset the branch. I think I'll open a new one when I reach some results.

**Comment by programarivm** [comment]
It seems as though this issue may not be too obvious, and I can't foresee its implications in specific terms at this moment. 

Probably a new TDD cycle can be started by adding sample data in `tests/unit/PGN/Validate/MovetextTest.php`.

```php
<?php

namespace Chess\Tests\Unit\PGN\Validate;

use Chess\PGN\Validate;
use Chess\Tests\AbstractUnitTestCase;

class MovetextTest extends AbstractUnitTestCase
{
    public static $validData = [
        '1.d4 Nf6 2.Nf3 e6 3.c4 Bb4+ 4.Nbd2 O-O 5.a3 Be7 6.e4 d6 7.Bd3 c5',
        '1.d4 Nf6 2.Nf3 e6 3.c4 Bb4+ 4.Nbd2 0-0 5.a3 Be7 6.e4 d6 7.Bd3 c5', // FIDE castling
        '1.e4 Nf6 2.e5 Nd5 3.d4 d6 4.Nf3 dxe5 5.Nxe5 c6 6.Be2 Bf5 7.c3 Nd7',
        '1.e4 c5 2.Nf3 Nc6 3.d4 cxd4 4.Nxd4 Nf6 5.Nc3 e5 6.Ndb5 d6 7.Bg5 a6 8.Na3',
        '1.d4 Nf6 2.c4 e6 3.Nc3 Bb4 4.e3 O-O 5.a3 Bxc3+ 6.bxc3 b6 7.Bd3 Bb7 8.f3 c5',
        '1.d4 Nf6 2.c4 e6 3.Nc3 Bb4 4.e3 0-0 5.a3 Bxc3+ 6.bxc3 b6 7.Bd3 Bb7 8.f3 c5', // FIDE castling
        '1.Nf3 Nf6 2.c4 c5 3.g3 b6 4.Bg2 Bb7 5.O-O e6 6.Nc3 a6 7.d4 cxd4 8.Qxd4 d6',
        '1.Nf3 Nf6 2.c4 c5 3.g3 b6 4.Bg2 Bb7 5.0-0 e6 6.Nc3 a6 7.d4 cxd4 8.Qxd4 d6', // FIDE castling
    ];

    /**
     * @dataProvider validData
     * @test
     */
    public function valid($movetext)
    {
        $this->assertSame($movetext, Validate::movetext($movetext));
    }

    ...
}
``` 
This way the PHP stack trace will inform about the files involved in the refactoring.

**Pull request closed without merging**